tests/providers/microsoft/azure/hooks/test_azure_storage_blob.py

```diff
@@ -0,0 +1,293 @@
+
+    @mock.patch("airflow.providers.microsoft.azure.hooks.azure_storage_blob.BlobServiceClient")
+    def test_create_container(self, mock_service):
+        hook = AzureStorageBlobHook(azure_blob_conn_id=self.shared_key_conn_id)
+        hook.create_container(container_name='mycontainer')
+        mock_service.return_value.get_container_client.assert_called_once_with('mycontainer')
+        mock_service.return_value.get_container_client.return_value.create_container.\
+            assert_called()
+
+    @mock.patch("airflow.providers.microsoft.azure.hooks.azure_storage_blob.BlobServiceClient")
+    def test_delete_container(self, mock_service):
+        hook = AzureStorageBlobHook(azure_blob_conn_id=self.shared_key_conn_id)
+        hook.delete_container('mycontainer')
+        mock_service.return_value.get_container_client.assert_called_once_with('mycontainer')
+        mock_service.return_value.get_container_client.return_value.delete_container. \
+            assert_called()
+
+    @mock.patch("airflow.providers.microsoft.azure.hooks.azure_storage_blob.BlobServiceClient")
+    def test_list(self, mock_service):
+        hook = AzureStorageBlobHook(azure_blob_conn_id=self.shared_key_conn_id)
+        hook.list(container_name='mycontainer',
+                  name_starts_with='my',
+                  include=None,
+                  delimiter='/')
+        mock_service.return_value.get_container_client.assert_called_once_with('mycontainer')
+        mock_service.return_value.get_container_client.return_value.walk_blobs.\
+            assert_called_once_with(name_starts_with='my',
+                                    include=None,
+                                    delimiter='/')
+
+    @mock.patch("airflow.providers.microsoft.azure.hooks.azure_storage_blob.BlobServiceClient")
+    def test_upload(self, mock_service):
+        hook = AzureStorageBlobHook(azure_blob_conn_id=self.shared_key_conn_id)
+        hook.upload(container_name='mycontainer',
+                    blob_name='myblob',
+                    data=b'mydata',
+                    blob_type='BlockBlob',
+                    length=4)
+        mock_service.return_value.get_container_client.assert_called_once_with('mycontainer')
+        mock_service.return_value.get_container_client.return_value.get_blob_client.\
+            assert_called_once_with('myblob')
+        mock_service.return_value.get_container_client.return_value.get_blob_client.\
+            return_value.upload_blob.assert_called_once_with(
+                b'mydata',
+                'BlockBlob',
+                length=4
+            )
+
+    @mock.patch("airflow.providers.microsoft.azure.hooks.azure_storage_blob.BlobServiceClient")
+    def test_download(self, mock_service):
+        hook = AzureStorageBlobHook(azure_blob_conn_id=self.shared_key_conn_id)
+        container_client = mock_service.return_value.get_container_client.return_value
+        blob_client = container_client.get_blob_client.return_value
```

Comment:
You need to add a return value to both of these

Reply:
Ok. Thanks.